src/sonic-config-engine/minigraph.py

```diff
@@ -323,6 +323,17 @@ def parse_meta(meta, hname):
                     deployment_id = value
     return syslog_servers, dhcp_servers, ntp_servers, mgmt_routes, erspan_dst, deployment_id
 
+def parse_deviceinfo(meta, hname):
+    ethernet_interfaces = []
+
+    device_info = meta.find(str(QName(ns, "DeviceInfo")))
```

Comment:
In DeviceInfos, there are many deviceinfo. You need findall deviceinfo. and for each deviceinfo, you need to find the hwsku and only choose the hwsku that is same as hwsku.